consensus/safety-rules/src/tests/local.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{tests::suite, InMemoryStorage, SafetyRulesManager, TSafetyRules};
+use consensus_types::common::{Payload, Round};
+use libra_types::crypto_proxies::ValidatorSigner;
+use std::sync::Arc;
+
+#[test]
+fn test() {
+    suite::run_test_suite(safety_rules::<Round>, safety_rules::<Vec<u8>>);
+}
+
+fn safety_rules<T: Payload>() -> (Box<dyn TSafetyRules<T>>, Arc<ValidatorSigner>) {
```

Comment:
Also, if we can’t for whatever reason, then please try to avoid sticking it in an Arc, that’s even worse :P

Reply:
I'll investigate the generics once things stabilize. But this begs the question, why not Arc<RwLock<Box< ;)

Reply:
The issue will specifically come if you want to have your API be async ever, since you can't await on something that is inside of a Mutex due to MutexGuard being !Send